src/Microsoft.CodeAnalysis.Analyzers/VisualBasic/MetaAnalyzers/Fixers/BasicPreferIsKindFix.vb

```diff
@@ -0,0 +1,59 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers
+Imports Microsoft.CodeAnalysis.CodeFixes
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Analyzers.MetaAnalyzers.CodeFixes
+    <ExportCodeFixProvider(LanguageNames.VisualBasic, Name:=NameOf(BasicPreferIsKindFix))>
+    <[Shared]>
+    Public NotInheritable Class BasicPreferIsKindFix
+        Inherits PreferIsKindFix
+
+        Protected Overrides Async Function ConvertKindToIsKindAsync(document As Document, sourceSpan As TextSpan, cancellationToken As CancellationToken) As Task(Of Document)
```

Comment:
Consider, if/when appropriate, extracting as much as possible in the base class.

Reply:
➡️ Also requested in https://github.com/dotnet/roslyn-analyzers/pull/3537#discussion_r555456764. I'm not a fan of `SyntaxGenerator` compared to `SyntaxFactory` due to its lack of fidelity, so there isn't much code eligible for sharing at this point.